src/components/workflowEditor/workflowEditor.tsx

```diff
@@ -364,6 +387,87 @@ class WorkflowEdit extends Component<WorkflowEditProps, WorkflowEditState> {
         return ciNode?.downstreams?.length || 0
     }
 
+    handleClearSelectedNode = () => {
+        this.handleSelectedNodeChange(null)
+    }
+
+    // TODO: Look into why have to parse id into String in few cases
+    handleSelectedNodeChange = (selectedNode: SelectedNode) => {
+        // If selectedNode is null, then remove bufferNodes
+        // else find the workflow in which the selectedNode is present and add bufferNodes
+        // bufferNodes are nodes which are have y greater than the selectedNode and are not present in downstream of selectedNode and the subsequent downstreams of the downstreams.
+        // So basically they are nodes in which we have to add buffer height
+        // and compute the maximum y of the bufferNodes and if there are downstreams > 1 then also for maxY, we will check maxY using node.y and depth
+        if (selectedNode) {
+            // would check if selectedNode.type and selectedNode.id is same as node.id and node.type
+            const _wf = this.state.workflows.find((wf) =>
+                wf.nodes.find(
+                    (wfNode) => String(wfNode.id) === String(selectedNode.id) && wfNode.type === selectedNode.nodeType,
+                ),
+            )
+            const _node = _wf?.nodes.find(
```

Comment:
this can be found in above iteration itself.

Reply:
Done